bokehjs/src/lib/models/widgets/date_picker.ts

```diff
@@ -39,7 +39,11 @@ export class DatePickerView extends InputWidgetView {
     super.connect_signals()
 
     const {value, min_date, max_date, disabled_dates, enabled_dates, position, inline} = this.model.properties
-    this.connect(value.change, () => this._set("defaultDate", value.value()))
+    this.connect(value.change, () => {
+      if (this._picker != null) {
+        this._picker.setDate(value.value())
+      }
```

Comment:
This could be simplified with modern ES syntax:
```ts
this._picker?.setDate(value.value())
```

Reply:
Would it make sense to replace the `_set` private method as well?

Reply:
I would drop `_set()` completely, especially given it is type unsafe (e.g. uses `string` key)